### PR TITLE
Remove death watch and supervision mechanics

### PR DESCRIPTION
**Pull request by chbatey**
These were missed in #24176

Also some removal of unused imports and a couple compiler warnings

### CODE REVIEW

File: akka-actor-typed/src/main/scala/akka/actor/typed/ActorSystem.scala [hunk 1]
```diff
@@ -139,8 +139,7 @@ abstract class ActorSystem[-T] extends ActorRef[T] with Extensions {
    * Ask the system guardian of this system to create an actor from the given
    * behavior and props and with the given name. The name does not need to
    * be unique since the guardian will prefix it with a running number when
-   * creating the child actor. The timeout sets the timeout used for the [[akka.actor.typed.scaladsl.AskPattern]]
-   * invocation when asking the guardian.
+   * creating the child actor.
```
Comment (johanandren) [inline akka-actor-typed/src/main/scala/akka/actor/typed/ActorSystem.scala]: But the timeout is still there?

**Review by johanandren — APPROVED** [review]

**Comment by chbatey** [comment]
It is but it’s kind of fake as the adapted system doesn’t need to wait to
spawn a system actor only the native one did so thought I’d remove it from
the docs

File: akka-actor-typed/src/main/scala/akka/actor/typed/internal/ActorContextImpl.scala [hunk 2]
```diff
@@ -57,10 +57,10 @@ import scala.concurrent.ExecutionContextExecutor
     internalSpawnAdapter(f, "")
 
   override def spawnAdapter[U](f: java.util.function.Function[U, T]): akka.actor.typed.ActorRef[U] =
-    internalSpawnAdapter(f.apply _, "")
+    internalSpawnAdapter(f.apply, "")
```
Comment (ktoso) [inline akka-actor-typed/src/main/scala/akka/actor/typed/internal/ActorContextImpl.scala]: isn't the `_` to avoid compiler warnings though?

**Review by ktoso — APPROVED** [review]
LGTM AFAICS

File: akka-actor-typed/src/main/scala/akka/actor/typed/internal/ActorContextImpl.scala (hunk 2 above)
Comment (chbatey) [reply akka-actor-typed/src/main/scala/akka/actor/typed/internal/ActorContextImpl.scala]: Dont think so, there aren't any compiler warnings in the typed module

Comment (ktoso) [reply akka-actor-typed/src/main/scala/akka/actor/typed/internal/ActorContextImpl.scala]: Ok, cool then 👍

**Pull request merged**